lib/remote/objectqueryhandler.cpp

```diff
@@ -194,14 +202,31 @@ bool ObjectQueryHandler::HandleRequest(
 	std::unordered_map<Type*, std::pair<bool, std::unique_ptr<Expression>>> typePermissions;
 	std::unordered_map<Object*, bool> objectAccessAllowed;
 
-	for (ConfigObject::Ptr obj : objs) {
+	response.result(http::status::ok);
+	response.set(http::field::content_type, "application/json");
+
+	auto adapter = std::make_shared<decltype(BeastHttpMessageAdapter(response))>(response);
+	JsonEncoder encoder(adapter, request.GetLastParameter("pretty"));
+
+	auto it = objs.begin();
+	auto generatorFunc = [&]() -> std::optional<Value> {
+		if (it == objs.end()) {
+			return std::nullopt;
+		}
+
+		if (it != objs.begin()) {
+			response.Write(yc);
+		}
```

Comment:
Isn't the callback function defined here called indirectly by some of the recursive encode calls there?

https://github.com/Icinga/icinga2/blob/fa99c024830720585db83848b56447b43573edfd/lib/base/json.hpp#L198-L210

If so, this would introduce the "async operation while a mutex is held problems" (unlock from another thread is undefined behavior, and a slow client delays the unlock operation) again.